File: alfa-beto-api/src/main/java/com/alfabetoapi/controller/response/ExerciseAttemptResponse.java
package com.alfabetoapi.controller.response;

import com.alfabetoapi.model.enums.ExerciseDifficultyEnum;
import com.alfabetoapi.model.enums.ExerciseTypeEnum;
import lombok.Builder;
import lombok.Data;

import java.math.BigInteger;

@Data
@Builder
public class ExerciseAttemptResponse {
    private Long exerciseId;
    private String exerciseName;
    private ExerciseTypeEnum type;
    private ExerciseDifficultyEnum difficulty;
    private BigInteger rewardCoins;
    private Long errorsPermitted;
    private Long exerciseAttemptId;
}

File: alfa-beto-app/src/constants/models/response/exercise-attempt-response.ts
import { EXERCISE_TYPE, EXERCISE_DIFFICULTY } from "../.."

export type ExerciseAttemptResponse = {
  exerciseId: number
  exerciseName: string
  type: EXERCISE_TYPE
  difficulty: string
  rewardCoins: number
  errorsPermitted: number
  exerciseAttemptId: number
}
